src/backends/fixerio/index.js

```diff
@@ -114,14 +118,21 @@ class FixerIoBackend {
     //
     let rate = new BigNumber(destinationRate).div(sourceRate)
     rate = this._subtractSpread(rate)
-
-    const sourceScale = this.getInfo(params.source_ledger).currencyScale
-    const destinationScale = this.getInfo(params.destination_ledger).currencyScale
-    const sourceAmount = new BigNumber(PROBE_SOURCE_AMOUNT).shift(sourceScale)
-    const destinationAmount = new BigNumber(sourceAmount)
-      .times(rate)
-      .shift(destinationScale - sourceScale)
-    return { points: [[0, 0], [sourceAmount.toNumber(), destinationAmount.toNumber()]] }
+      .shift(destinationInfo.currencyScale - sourceInfo.currencyScale)
+
+    return (sourceInfo.maxBalance === undefined
```

Comment:
getCurve is a generator, so can you restructure this to use yield, instead of returning a promise? As it's written here, yielding getCurve would return a promise, effectively making getCurve return a promise to a promise